src/systems/a320_systems/src/electrical/mod.rs

```diff
@@ -2191,18 +2193,51 @@ mod a320_electrical_circuit_tests {
     }
 
     struct TestHydraulicSystem {
-        is_rat_hydraulic_loop_pressurised: bool,
+        time_since_start: Duration,
+        starting_or_started: bool,
+        emergency_motor_speed: AngularVelocity,
     }
     impl TestHydraulicSystem {
-        fn new(is_rat_hydraulic_loop_pressurised: bool) -> Self {
+        fn new() -> Self {
             Self {
-                is_rat_hydraulic_loop_pressurised,
+                time_since_start: Duration::from_secs(0),
+                starting_or_started: false,
+                emergency_motor_speed: AngularVelocity::new::<revolution_per_minute>(0.),
+            }
+        }
+
+        fn update(
+            &mut self,
+            is_rat_hydraulic_loop_pressurised: bool,
+            is_emergency_elec: bool,
+            rat_man_on_pressed: bool,
+            context: &UpdateContext,
+        ) {
+            if (is_emergency_elec || rat_man_on_pressed) && is_rat_hydraulic_loop_pressurised {
+                self.starting_or_started = true;
+            }
+
+            if self.starting_or_started {
+                self.time_since_start += context.delta();
+            }
+
+            if self.time_since_start > Duration::from_secs(8) && is_rat_hydraulic_loop_pressurised {
+                self.emergency_motor_speed = AngularVelocity::new::<revolution_per_minute>(12000.);
+            } else {
+                self.emergency_motor_speed = AngularVelocity::new::<revolution_per_minute>(0.);
             }
         }
     }
-    impl RamAirTurbineHydraulicCircuitPressurised for TestHydraulicSystem {
-        fn is_rat_hydraulic_circuit_pressurised(&self) -> bool {
-            self.is_rat_hydraulic_loop_pressurised
+    impl GeneratorControlUnitInterface for TestHydraulicSystem {
```

Comment:
**suggestion**: In case I didn't mention this elsewhere. Think of an abstraction that removes `...Interface` from the name of the trait.

Reply:
Should be done :)